utils/text_extractor.py

```diff
@@ -24,15 +25,15 @@ def findTitle(html):
 
 
 def get_all_text_from_url(url, retries=5):
-    if retries == 0:
+    if retries < 1:
         logging.error("get_all_text_from_url: could not handle request")
         raise Exception("could not handle request")
     try:
         url_utf_8 = safe_url_string(url)
         headers = {'User-Agent': USER_AGENT}
         request = urllib.request.Request(url_utf_8, None, headers)
 
-        response = urllib.request.urlopen(request)
+        response = urllib.request.urlopen(request, timeout=current_app.config['URL_TIMEOUT'])
         given_encoding = response.headers.get_content_charset()
```

Comment:
Extract to the constant in the beginning of a file, so it will be easier to maintain and be more readable